### ISSUE TITLE
NDCG and Tied Scores

### ISSUE THREAD

**Issue by miguellacerda** [microsoft/LightGBM#3363]
Hi there

_How do you handle tied scores when calculating the NDCG?_

I am checking if I can reproduce the value of the NDCG eval metric reported by LGBMRanker. To do this for a given query, I use the fitted model's predict method to obtain the predicted scores on my validation set, sort the items by these scores and then sum up the discounted gain for the top _T_ scores. However, several items get the same predicted score so their ranking is ambiguous. How does this get resolved?

Thanks!

**Comment by guolinke** [comment]
@miguellacerda I remember that we will use the higher score when some pairs are tied.
